applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/servers.js

```diff
@@ -1,21 +1,61 @@
 'use strict';
 'require form';
+'require uci';
+'require ui';
 'require shadowsocks-libev as ss';
 
-function startsWith(str, search) {
-	return str.substring(0, search.length) === search;
-}
+var conf = 'shadowsocks-libev';
 
 return L.view.extend({
 	render: function() {
 		var m, s, o;
 
-		m = new form.Map('shadowsocks-libev', _('Remote Servers'),
+		m = new form.Map(conf, _('Remote Servers'),
 			_('Definition of remote shadowsocks servers.  \
 				Disable any of them will also disable instances referring to it.'));
 
 		s = m.section(form.GridSection, 'server');
 		s.addremove = true;
+		s.handleLinkImport = function() {
+			var textarea = new ui.Textarea();
+			ui.showModal(_('Import Links'), [
+				textarea.render(),
+				E('div', { class: 'right' }, [
+					E('button', {
+						class: 'btn',
+						click: ui.hideModal
+					}, [ _('Cancel') ]),
+					' ',
+					E('button', {
+						class: 'btn cbi-button-action',
+						click: ui.createHandlerFn(this, function() {
+							textarea.getValue().split('\n').forEach(function(s) {
+								var config = ss.parse_uri(s);
+								if (config) {
+									var tag = config[1];
+									if (tag && !tag.match(/^[a-zA-Z0-9_]+$/)) tag = null;
+									var sid = uci.add(conf, 'server', tag);
+									config = config[0];
+									Object.keys(config).forEach(function(k) {
+										uci.set(conf, sid, k, config[k]);
+									});
+								}
+							});
+							return this.handleModalSave(uci);
```

Comment:
I found a tricky way. Because `handleModalSave` calls `.save()` on first argument, passing `uci` works.
https://github.com/openwrt/luci/blob/6496ed88a6dfa15c0f34f3865a559e1c405cdc4f/modules/luci-base/htdocs/luci-static/resources/form.js#L1313-L1319
Or should I instead copy paste the code of `handleModalSave` here?

Reply:
It looks a hack to me ;)  I prefer we make it clear by duplicating part of the logic from `CBITableSection.handleModalSave()`.  Let's hear @jow- 's opinion first.

The other thing is that it seems to me height of the textarea element cannot be resized (Chrome 77.0.3865.120).  It would be great if there is a way around this in the context of LuCI framework.

Reply:
What about `this.map.handleModalSave(this.map, ev);` ?

Reply:
There's no `handleModalSave` function in `this.map`.

Reply:
Okay, and `form.TableSection.prototype.handleModalSave.call(this, this.map, ev);` ?

Reply:
Imported data were lost. I think `uci.save()` is essential.

Reply:
Ping @jow-

Reply:
Ah right, the explicit uci save is needed because you create the data outside of the map scope. In this please use `return uci.save().then(L.bind(this.handleModalSave, this));`

Reply:
No working. Because `handleModalSave` takes a `modalMap` argument, result in calling `undefined.save()`. If using `L.bind(this.handleModalSave, this, this.map)`, the config get deleted.
![image](https://user-images.githubusercontent.com/12028138/68791088-8fb97580-0683-11ea-98c5-179dce6403b0.png)